src/NLog/Config/PropertyTypeConverter.cs

```diff
@@ -43,28 +44,40 @@ class PropertyTypeConverter : IPropertyTypeConverter
         /// <inheritdoc/>
         public object Convert(object propertyValue, Type propertyType, string format, IFormatProvider formatProvider)
         {
-            if (propertyType == null || propertyValue == null || propertyValue.GetType() == propertyType || propertyType == typeof(object))
+            if (!NeedToConvert(propertyValue, propertyType))
             {
                 return propertyValue;
             }
 
+            bool? isNullable = null;
+            Type nullableType = null;
```

Comment:
Should just be `Type nullableType = Nullable.GetUnderlyingType(propertyType);`, since you are doing the lookup no matter what. (Then you can also remove `bool? isNullable `).

Reply:
indeed! Thanks!

Reply:
Could consider having a conversion-dictionary. Where common types are registered along with their converter. Similar to what has been implemented in PropertyHelper. The conversion dictionary could have the type `DateTime`, `DateTime?`, `Guid`, `Guid?`, `int`, `int?`, `bool`, `bool?`, `string`. This would make the conversion path quicker.